static/app/components/tooltip.tsx

```diff
@@ -116,215 +111,262 @@ function computeOriginFromArrow(
   }
 }
 
-class Tooltip extends React.Component<Props, State> {
-  static defaultProps: DefaultProps = {
-    position: 'top',
-    containerDisplayMode: 'inline-block',
-  };
-
-  state: State = {
-    isOpen: false,
-    usesGlobalPortal: true,
-  };
-
-  async componentDidMount() {
-    if (IS_ACCEPTANCE_TEST) {
-      const TooltipStore = (await import('sentry/stores/tooltipStore')).default;
-      TooltipStore.addTooltip(this);
+function getTooltipPortal(useGlobalPortal: boolean): HTMLElement {
+  // Use a single global portal element for tooltips
+  if (useGlobalPortal) {
+    let portal = document.getElementById('tooltip-portal');
+    if (!portal) {
+      portal = document.createElement('div');
+      portal.setAttribute('id', 'tooltip-portal');
+      document.body.appendChild(portal);
     }
+    return portal;
   }
 
-  async componentWillUnmount() {
-    const {usesGlobalPortal} = this.state;
+  // Each tooltip gets it's own div element
+  const portal = document.createElement('div');
+  document.body.appendChild(portal);
+  return portal;
+}
 
-    if (IS_ACCEPTANCE_TEST) {
-      const TooltipStore = (await import('sentry/stores/tooltipStore')).default;
-      TooltipStore.removeTooltip(this);
-    }
-    if (!usesGlobalPortal) {
-      document.body.removeChild(this.getPortal(usesGlobalPortal));
-    }
+/**
+ * Codesplit loading of the TooltipStore used for visual acceptance tests
+ */
+async function getAcceptanceTestTooltipStore() {
```

Comment:
Extracted all the logic for acceptance test usage into this hook. Hook separation of concerns ftw.